src/sdk/accountspace_nb.js

```diff
@@ -217,39 +212,148 @@ class AccountSpaceNB {
     /////////////////////////////////
 
     async create_access_key(params, account_sdk) {
-        // TODO
-        dbg.log0('AccountSpaceNB.create_access_key:', params);
-        const { code, http_code, type } = IamError.NotImplemented;
-        throw new IamError({ code, message: 'NotImplemented', http_code, type });
+        const action = IAM_ACTIONS.CREATE_ACCESS_KEY;
+        const requesting_account = system_store.get_account_by_email(account_sdk.requesting_account.email);
+        const account_name = account_util.populate_username(params.username, requesting_account.name.unwrap());
+        account_util._check_if_account_exists(action, account_name);
+        const requested_account = system_store.get_account_by_email(account_name);
+        account_util._check_if_requesting_account_is_root_account(action, requesting_account, { username: params.username });
+        account_util._check_if_requested_account_is_root_account_or_IAM_user(action, requesting_account, requested_account);
+        account_util._check_if_requested_is_owned_by_root_account(action, requesting_account, requested_account);
+        account_util._check_number_of_access_key_array(action, requested_account);
+        const req = {
+            rpc_params: {
+                email: account_name,
+                is_iam: true,
+            },
+            account: requesting_account,
+        };
+        // CORE CHANGES PENDING - START
+        let iam_access_key;
+        try {
+            iam_access_key = await account_util.generate_account_keys(req);
+        } catch (err) {
+            dbg.error(`AccountSpaceNB.${action} error: `, err);
+            const message_with_details = `Create accesskey failed for the user with name ${params.username}.`;
+            const { code, http_code, type } = IamError.InternalFailure;
+            throw new IamError({ code, message: message_with_details, http_code, type });
+        }
+
+        // CORE CHANGES PENDING - STOP
+
+        return {
+            username: params.username,
```

Comment:
username is not required field, it can be on user on himself (link to docs - [CreateAccessKey](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateAccessKey.html)).